invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java

```diff
@@ -164,43 +164,47 @@ protected InvoiceApiException generateAlreadyExistsException(final InvoiceModelD
     }
 
     @Override
-    public List<InvoiceModelDao> getInvoicesByAccount(final Boolean includeVoidedInvoices, final InternalTenantContext context) {
+    public List<InvoiceModelDao> getInvoicesByAccount(final Boolean includeVoidedInvoices, final Boolean withInvoiceItems, final InternalTenantContext context) {
         final List<Tag> invoicesTags = getInvoicesTags(context);
 
         return transactionalSqlDao.execute(true, entitySqlDaoWrapperFactory -> {
             final InvoiceSqlDao invoiceSqlDao = entitySqlDaoWrapperFactory.become(InvoiceSqlDao.class);
             final List<InvoiceModelDao> invoicesByAccountRecordId = invoiceSqlDao.getByAccountRecordId(context);
             final List<InvoiceModelDao> invoices = invoicesByAccountRecordId.stream()
-                    .filter(invoice -> !invoice.isMigrated() &&
-                                       (includeVoidedInvoices || !InvoiceStatus.VOID.equals(invoice.getStatus())))
-                    .sorted(INVOICE_MODEL_DAO_COMPARATOR)
-                    .collect(Collectors.toUnmodifiableList());
+                                                                            .filter(invoice -> !invoice.isMigrated() &&
+                                                                                               (includeVoidedInvoices || !InvoiceStatus.VOID.equals(invoice.getStatus())))
+                                                                            .sorted(INVOICE_MODEL_DAO_COMPARATOR)
+                                                                            .collect(Collectors.toUnmodifiableList());
 
-            invoiceDaoHelper.populateChildren(invoices, invoicesTags, entitySqlDaoWrapperFactory, context);
+            if (withInvoiceItems) {
```

Comment:
The `populateChildren` does more than pulling the invoice items, so eliminating calling this based on the value of this flag seems incorrect (too restrictive). Are we trying to just return 'shell' invoices (i.e not populated), in which case we should change the name of this flag, or are we simply trying to not pull invoice items but keep everything else the way `populateChildren` does?

Reply:
Taking a look at the code again, I see what you mean. So rather than using the flag here, it should be passed to the `populateChildren` method and used at [Line#255](https://github.com/killbill/killbill/blob/35d35aeffc512fc2052b7c65aa686a3a3865b4ba/invoice/src/main/java/org/killbill/billing/invoice/dao/InvoiceDaoHelper.java#L255) to determine whether `setInvoiceItemsWithinTransaction` should be invoked or not. Does that sound right?

Reply:
If we want to really honor what this flag says, then yes. But more importantly, what is the behavior we really want? 
/cc @pierre Do we implement a way to retrieve shell invoices (i.e we don't populate anything), or we simply don't pull the invoice items? 

My thoughts: I think we really want the former.

Reply:
> Do we implement a way to retrieve shell invoices (i.e we don't populate anything)

Yup, that's what we want. Pulling anything else and/or computing amount/balance/etc. will be a performance hit.

Skipping `invoiceDaoHelper.populateChildren` as it is now looks like the right approach to me.

Reply:
Left the behavior as it is and renamed the flag to `includeInvoiceComponents`